Realm/RLMUser.h

```diff
@@ -144,6 +163,42 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (RLMRealmConfiguration *)flexibleSyncConfiguration NS_REFINED_FOR_SWIFT;
 
+/**
+ Create a flexible sync configuration instance, which can be used to open a Realm that
+ supports flexible sync.
+ 
+ It won't possible to combine flexible and partition sync in the same app, which means if you open
+ a realm with a flexible sync configuration, you won't be able to open a realm with a PBS configuration
+ and the other way around.
```

Comment:
"won't" is the wrong word here.

This paragraph is confusing because of how we've overloaded the word "app". You can use both PBS and FLX in the same app and they just need different server-side apps. It's also misleading because opening a Realm with a flexible sync configuration isn't what makes it invalid to open that same realm in partition-based mode.